WordPress/gutenberg
# Style engine: add handling of spacing presets

## What?
Adds handling of conversion of spacing preset values to css vars to style engine

## Why?
#41527 adds the spacing presets to theme.json, as the first part of implementing standardised design tokens for spacing as detailed in #39371

## How?
Extends the existing preset var handling added for the processing of color presets.

## Testing Instructions

- Add the following block markup in the edit code view and save and view in the frontend:

```

```

- Check in the frontend that the block markup has converted the `var:preset|space|20` values into `var(--wp--preset--spacing--20);` in the block styles

**N.B.** The frontend styles markup in this PR is not an indication of the final marking for spacing presets. The final implementation may use utility classes as noted in #39371, but the fact that we are generating these styles dynamically on the frontend means we can modify the output relatively easily as we go.

## Screenshots or screencast

<img width="1095" alt="Screen Shot 2022-06-14 at 2 42 07 PM" src="https://user-images.githubusercontent.com/3629020/173496060-371f1a8d-8228-4d4f-b984-880e08420bc1.png">
